src/isomorphic/classic/element/ReactElement.js

```diff
@@ -243,10 +243,19 @@ ReactElement.cloneElement = function(element, config, children) {
       key = '' + config.key;
     }
     // Remaining properties override existing props
+    var defaultProps = {};
```

Comment:
We are doing an extra/unnecessary allocation here, which creates extra gc pressure.  See below.